app/handlers/testConfig.py

```diff
@@ -29,79 +29,121 @@ class TestConfigHandler(tornado.web.RequestHandler):
+    def test_object_callback(status_code, response_message, identifier):
+        print(status_code, response_message)
+
+
+class TestConfigFeatureThread(threading.Thread):
+    def __init__(self, identifier, frame_start, num_frames, callback):
+        threading.Thread.__init__(self)
+        self.identifier = identifier
+        self.frame_start = frame_start
+        self.num_frames = num_frames
+        self.callback = callback
+
+    def run(self):
+        project_path = get_project_path(self.identifier)
         tracking_path = os.path.join(project_path, "tracking.cfg")
         db_path = os.path.join(project_path, ".temp", "test", "test_feature", "test1.sqlite")
         if os.path.exists(db_path):
             os.remove(db_path)
 
-        testing_dict = {'frame1': frame_start, 'nframes': num_frames}
+        testing_dict = {'frame1': self.frame_start, 'nframes': self.num_frames}
         update_config_without_sections(tracking_path, testing_dict)
 
-        images_folder = "feature_images"
-        video.delete_files(images_folder)        
+        images_folder = os.path.join(get_project_path(self.identifier), "feature_images")
+        video.delete_files(images_folder)
 
         try:
             subprocess.call(["feature-based-tracking", tracking_path, "--tf", "--database-filename", db_path])
-            subprocess.call(["display-trajectories.py", "-i", get_project_video_path(identifier), "-d", db_path, "-o", project_path + "/homography/homography.txt", "-t", "feature", "--save-images", "-f", str(frame_start), "--last-frame", str(frame_start+num_frames)])
+            subprocess.call(["display-trajectories.py", "-i", get_project_video_path(self.identifier), "-d", db_path, "-o", project_path + "/homography/homography.txt", "-t", "feature", "--save-images", "-f", str(self.frame_start), "--last-frame", str(self.frame_start+self.num_frames)])
         except Exception as err_msg:
-            return (500, err_msg)
-
-        video.move_files_to_folder(os.getcwd(),images_folder,'image-', '.png')
-
-        return (200, "Success")
-
-    def runConfigTestObject(self, identifier, frame_start, num_frames):
-        project_path = get_project_path(identifier)
-        if not os.path.exists(project_path):
-           return (500, 'Project directory does not exist. Check your identifier?')
-
-
+            return self.callback(500, err_msg, self.identifier)
+
+        videos_folder = os.path.join(get_project_path(self.identifier), "feature_images")
+        video_filename = "feature_video.mp4"
+        temp_image_prefix = 'image-'
+        video.move_files_to_folder(os.getcwd(),images_folder,temp_image_prefix, '.png')
+        video.renumber_frames(images_folder, 0, temp_image_prefix, "png")
+        video.convert_frames_to_video(get_project_video_path(self.identifier), images_folder, videos_folder, temp_image_prefix, video_filename, 1.0)
+        video.delete_files(images_folder, prefix=temp_image_prefix, extensions=["png"])
```

Comment:
Could it be possible to merge all these into a single helper method that takes in a video filename?